htmlescape.cc

```diff
@@ -4,11 +4,7 @@
 #include <QString>
 #include <QTextDocumentFragment>
 
-#if QT_VERSION >= QT_VERSION_CHECK( 5, 0, 0 )
 #include <QRegularExpression>
-#else
-#include <QRegExp>
-#endif
```

Comment:
Did you use automatic tools to make changes like this or made them manually?

I mean: `QT_VERSION_CHECK($x, $y, $z)` can be replaced with the macro's expanding value using some regular expression tool as the first step. Then [unifdef](https://dotat.at/prog/unifdef/) or [Coan](http://coan2.sourceforge.net/) can clean up no longer needed code in the second step.

If the process can be automated, the correctness would be much simpler for reviewers to verify as they could just run the same commands themselves and compare the results. If this works, the automated change should be in the first commit of this pull request; or better yet - in the last commit, in which case it could even be split into another PR to simplify the review of this one.

Another benefit of automating this cleanup is that we could rethink the minimum Qt version we should support. For instance, @embar- has showed interest in Qt 5.12 in a comment to #1400:
> For me – for use with Qt WebEngines – I am happy with git revision as of 20220104 (and still with some rejected changes related with merely porting for Qt 5.15, because I still use Qt 5.12 that also supports Qt WebEngine). I even  would vote to make commits related to additional moving to Qt 5.15 separately from WebEngines.

Reply:
qwebengine built with qt>5 which means all the QT_VERSION_CHECK<5 are useless .
I do not agree with you in this place.
QT5.12>5

Reply:
revert as many as I can .

Reply:
> qwebengine built with qt>5.4 which means all the QT_VERSION_CHECK<5 are useless .

useless != harmful. On the other hand, the extra diff is certainly a review complication.

> I do not agree with you in this place.

You don't agree that automation is useful? Do you believe that manual cleanup is somehow superior? If you do the automatic cleanup, you can compare the results with your manual changes. Who knows, maybe you'll even find a mistake in your manual cleanup.